Syndemics/R/internalTools_crc.R

```diff
@@ -0,0 +1,122 @@
+          all_formulas <- c(all_formulas, paste0(y, "~", paste(combination_additive, collapse="+"), "+", interaction_formula))
+        }
+      }
+    }
+  }
+
+  return(lapply(unique(all_formulas), as.formula))
+}
+
+#' Internal function to create a formula based on correlation among databases listed in \code{CRC}
+#'
+#' @param corr_matrix A correlation matrix
+#' @param threshold Threshold for interacting two databases
+#' @param freq.column frequency column, e.g. N_ID
+#'
+#' @keywords internal
+formula_corr <- function(corr_matrix, threshold, freq.column){
+  corr_upper <- corr_matrix; corr_upper[upper.tri(corr_matrix, diag = TRUE)] <- NA
+  interaction_indices <- which(abs(corr_upper) > threshold, arr.ind = TRUE)
+
+  interactions_df <- data.frame(var1 = rownames(corr_upper)[interaction_indices[, 1]],
+                                var2 = colnames(corr_upper)[interaction_indices[, 2]])
+
+  if (nrow(interactions_df) == 0) {
+    interactions_df <- data.frame(var1 = character(),
+                                  var2 = character(),
+                                  stringsAsFactors = FALSE)
+  }
+
+  interaction_terms <- apply(interactions_df, 1, function(row) {
+    paste0(row[1], "*", row[2])
+  })
+
+  non_interaction_vars <- setdiff(rownames(corr_matrix), c(interactions_df$var1, interactions_df$var2))
+  formula_string <- paste(paste(non_interaction_vars, collapse = " + "), "+", paste(interaction_terms, collapse = " + "))
+  formula_string <- paste(freq.column, "~", formula_string)
+
+  formula_object <- as.formula(formula_string)
+
+  return(formula_object)
+}
+
+#' Internal function to create a formula based on correlation among databases listed in \code{CRC}
+#'
+#' @param corr_matrix A correlation matrix
+#' @param threshold Threshold for interacting two databases
+#' @param freq.column frequency column, e.g. N_ID
+#'
+#' @keywords internal
+formula_corr <- function(corr_matrix, threshold, freq.column){
+  corr_upper <- corr_matrix; corr_upper[upper.tri(corr_matrix, diag = TRUE)] <- NA
+  interaction_indices <- which(abs(corr_upper) > threshold, arr.ind = TRUE)
+
+  interactions_df <- data.frame(var1 = rownames(corr_upper)[interaction_indices[, 1]],
+                                var2 = colnames(corr_upper)[interaction_indices[, 2]])
+
+  if (nrow(interactions_df) == 0) {
+    interactions_df <- data.frame(var1 = character(),
+                                  var2 = character(),
+                                  stringsAsFactors = FALSE)
+  }
+
+  interaction_terms <- apply(interactions_df, 1, function(row) {
+    paste0(row[1], "*", row[2])
+  })
+
+  non_interaction_vars <- setdiff(rownames(corr_matrix), c(interactions_df$var1, interactions_df$var2))
+  formula_string <- paste(paste(non_interaction_vars, collapse = " + "), "+", paste(interaction_terms, collapse = " + "))
+  formula_string <- paste(freq.column, "~", formula_string)
+
+  formula_object <- as.formula(formula_string)
+
+  return(formula_object)
+}
```

Comment:
Is there a difference between these lines and lines 58-89?

Reply:
Removed! Thanks for flagging